### DISCUSSION TITLE
Does a `keep_columns` method on `Dataset`, `IterableDataset`, etc. make sense?

### DISCUSSION THREAD

**Discussion by alvarobartt** [huggingface/datasets#5503] [Q&A] [answered]
I was dealing with a `Dataset` with way too many columns and I feel a bit tedious to do a `set(columns) - columns_to_keep` to then provide that as a list to `.remove_columns()`.

AFAIK this has been mentioned by other users in the past e.g. https://discuss.huggingface.co/t/how-to-create-a-new-dataset-from-another-dataset-and-select-specific-columns-and-the-data-along-with-the-column/15120, so does it make sense to add a method named `.keep_columns()` or something similar to do that, so that the user doesn't need to calculate the column diff on top of `.remove_columns()`.

I also understand that this may be a little over-engineering for such a small feature that maybe is not that common and in the end, it's just helping the developer with removing a single line of code, but I think it can help somehow. Anyway, I also get the counterpart engineering-wise.

So I'd appreciate your input here!

**Comment by mariosasko** [comment, marked as answer]
Hi! We are in the process of adding this method to the API (see https://github.com/huggingface/datasets/issues/5468).

**Reply by alvarobartt** [reply]
Hi, @mariosasko thanks for the follow-up, I'll close this Discussion when merged into `main` if you agree 🤗

**Reply by alvarobartt** [reply]
Thanks for that @mariosasko! To anyone else reading, PR at https://github.com/huggingface/datasets/pull/5480 was merged yesterday and is already available if you install `datasets` from source as `pip install git+https://github.com/huggingface/datasets@69b19755e9e37b746ef56780a62d21ef20c574d5`
